src/record/key.rs

```diff
@@ -217,6 +223,44 @@ impl RemoteId {
         full_id.push_str(alias.0.as_str());
         Self::new_unchecked(full_id, PROVIDER_LEN)
     }
+
+    /// Encode as a platform-friendly path into the provided buffer.
+    ///
+    /// In order to reduce the number of files which are in the same directory, we apply a 30-bit
+    /// header to each path, which is encoded in base32 as `xx/xx/xx`. Then the corresponding path
+    /// is
+    /// ```text
+    /// provider/xx/xx/xx/base32-encoding-of-sub-id
+    /// ```
+    /// The 30 bit header is formed by converting the u64 output of the
+    /// [highway](https://crates.io/crates/highway) hashing algorithm applied to the sub-id into
+    /// little endian bytes, then taking the four most significant bytes (in decreasing order),
+    /// encoding using BASE32 into 8 ASCII characters, and then taking the first 6.
+    ///
+    /// The header `xx/xx/xx` ensures that each directory does not have more than 1024 immediate
+    /// sub-directories.
+    pub fn encode_path(&self, path_buf: &mut PathBuf) {
```

Comment:
I don't feel this function belongs in `impl RemoteId` or even this file. We can have a new `path.rs` file maybe.